### ISSUE TITLE
Cannot mount haraka with jailmanage haraka after running provision haraka

### ISSUE THREAD

**Issue by barontel** [msimerson/Mail-Toaster-6#57]
After running provision haraka ....
cannot unmount '/jails/stage': Device busy
waiting for ZFS filesystem to quiet (0)
cannot unmount '/jails/stage': Device busy
waiting for ZFS filesystem to quiet (1)
cannot unmount '/jails/stage': Device busy
.
.
.
cannot unmount '/jails/stage': Device busy
waiting for ZFS filesystem to quiet (11)
service jail stop haraka
Stopping jails: haraka.
jail -r haraka
zroot/data/haraka filesystem exists
zroot/jails/haraka filesystem exists
zfs rename zroot/jails/haraka zroot/jails/haraka.last
cannot unmount '/jails/haraka': Device busy
waiting for ZFS filesystem to quiet (0)
.
.
.
cannot unmount '/jails/haraka': Device busy
trying to force rename (6)
waiting for ZFS filesystem to quiet (6)
zfs rename zroot/jails/haraka.ready zroot/jails/haraka

   ***   service jail start haraka   ***

Starting jails: cannot start jail  "haraka":
jail: haraka: mount: /jails/haraka/data/avg: No such file or directory
.

Success! A new 'haraka' jail is provisioned

# jailmanage haraka
Entering jail haraka
    /sbin/mount_nullfs /usr/ports /jails/haraka/usr/ports
    /sbin/mount_nullfs /var/cache/pkg /jails/haraka/var/cache/pkg
jexec: jail "haraka" is dying
jexec: jail "haraka" is dying
all done!
    /sbin/umount /jails/haraka/usr/ports
    /sbin/umount /jails/haraka/var/cache/pkg

Any idea how to fix this?

Thanks.

**Comment by barontel** [comment]
After writing above, ran provision avg. After avg as re-provisioned, ran:
 # jailmanage haraka
Entering jail haraka
    /sbin/mount_nullfs /usr/ports /jails/haraka/usr/ports
    /sbin/mount_nullfs /var/cache/pkg /jails/haraka/var/cache/pkg
jexec: jail "haraka" not found
jexec: jail "haraka" not found
all done!
    /sbin/umount /jails/haraka/usr/ports
    /sbin/umount /jails/haraka/var/cache/pkg

Strange because haraka is in /jails
 # ls
avg                     dovecot                 haraka-saved            rspamd                  webmail
avg.last                geoip                   haraka.last             rspamd.last             webmail.last
base-10.2-RELEASE       geoip.last              monitor                 spamassassin
clamav                  haproxy                 mysql                   spamassassin.last
dns                     haproxy.last            mysql.last              vpopmail
dns.last                haraka                  redis                   vpopmail.last

**Comment by barontel** [comment]
Ran all provisions after AVG to and including provision haraka
Seems haraka is built but still cannot mount it

Stopping jails:.
jail -r haraka
zroot/data/haraka filesystem exists
zroot/jails/haraka.last filesystem exists
zfs destroy zroot/jails/haraka.last
zroot/jails/haraka filesystem exists
zfs rename zroot/jails/haraka zroot/jails/haraka.last
zfs rename zroot/jails/haraka.ready zroot/jails/haraka

   ***   service jail start haraka   ***

Starting jails: cannot start jail  "haraka":
jail: haraka: mount: /jails/haraka/data/avg: No such file or directory
.

Success! A new 'haraka' jail is provisioned

**Comment by barontel** [comment]
Hi Matt,
I created a new haraka smtp-only VPS. 
Did everything exactly as your pages say.
**Installed everything perfectly**
# jls
   JID  IP Address      Hostname                      Path
     3  172.16.15.3     dns                           /jails/dns
     5  172.16.15.4     mysql                         /jails/mysql
     7  172.16.15.16    redis                         /jails/redis
     9  172.16.15.5     clamav                        /jails/clamav
    11  172.16.15.14    avg                           /jails/avg
    13  172.16.15.17    geoip                         /jails/geoip
    15  172.16.15.6     spamassassin                  /jails/spamassassin
    17  172.16.15.13    rspamd                        /jails/rspamd
    19  172.16.15.8     vpopmail                      /jails/vpopmail
    21  172.16.15.15    dovecot                       /jails/dovecot
    23  172.16.15.9     haraka                        /jails/haraka
    25  172.16.15.12    haproxy                       /jails/haproxy


Where is haraka??
# jailmanage haraka
[root@haraka /usr/local]# ls
bin/     etc/     include/ info/    lib/     libdata/ libexec/ man/     openssl/ sbin/    share/

**Comment by msimerson** [comment]
Looks to me like 23 is Haraka.

Also, consider deploying in a test environment first.

**Comment by toastedmail** [comment]
Matt, what does 23 mean?
To deploy in a test environ means setting up a vps, installing BSD et al.

**Comment by msimerson** [comment]
In your comment, you ask "Where is haraka??" Look up 3 lines at jail ID 23. What's its name? Looks to me like the answer to your question.

**Comment by barontel** [comment]
Hello Matt,
Yes I see '23 172.16.15.9 haraka /jails/haraka'

Tell me where haraka is when haraka jail is mounted:
root@smtpd:~ # jailmanage haraka
Entering jail haraka
    /sbin/mount_nullfs /usr/ports /jails/haraka/usr/ports
    /sbin/mount_nullfs /var/cache/pkg /jails/haraka/var/cache/pkg
curl-7.50.1 is vulnerable:
cURL -- Escape and unescape integer overflows
CVE: CVE-2016-7167
WWW: https://vuxml.FreeBSD.org/freebsd/b018121b-7a4b-11e6-bf52-b499baebfeaf.html

1 problem(s) in the installed packages found.
[root@haraka ~]# ls /usr/local/
bin/     etc/     include/ info/    lib/     libdata/ libexec/ man/     openssl/ sbin/    share/
[root@haraka ~]#

In our other server which has not yet been updated with the latest 'provision haraka', haraka exists:
root@mxbt1:~ # jailmanage haraka
Entering jail haraka
    /sbin/mount_nullfs /usr/ports /jails/haraka/usr/ports
    /sbin/mount_nullfs /var/cache/pkg /jails/haraka/var/cache/pkg
python27-2.7.11_1 is vulnerable:
Python -- Integer overflow in zipimport module
CVE: CVE-2016-5636
WWW: https://vuxml.FreeBSD.org/freebsd/1d0f6852-33d8-11e6-a671-60a44ce6887b.html

python27-2.7.11_1 is vulnerable:
Python -- smtplib StartTLS stripping vulnerability
CVE: CVE-2016-0772
WWW: https://vuxml.FreeBSD.org/freebsd/8d5368ef-40fe-11e6-b2ec-b499baebfeaf.html

node-5.2.0_1 is vulnerable:
node -- multiple vulnerabilities
CVE: CVE-2016-0797
CVE: CVE-2016-0705
CVE: CVE-2016-0702
WWW: https://vuxml.FreeBSD.org/freebsd/6d33b3e5-ea03-11e5-85be-14dae9d210b8.html

2 problem(s) in the installed packages found.
[root@haraka ~]# ls /usr/local/
bin/     etc/     haraka/  include/ info/    lib/     libdata/ man/     openssl/ sbin/    share/
[root@haraka ~]#

Do you now see what I have been trying to communicate to you?
Thanks Matt.

**Comment by msimerson** [comment]
The haraka data directory is located in /data/vpopmail, or just /data within the vpopmail jail.

**Comment by toastedmail** [comment]
Hi Matt,
Prior to the update on the github page Sept 17, 2016, provision haraka would install haraka in /jails/haraka/usr/local/haraka,

You changed Updating by adding mail-toaster.conf. Presumably to make it easier for people after they edit the few lines for future updates.

On our original smtpd server, when we ran the following:
sudo sh
. mail-toaster.sh
provision host
provision base

followed by **provision haraka**, Haraka jail could no longer be mounted as shown:

root@smtpd:~ # jailmanage haraka
Entering jail haraka
    /sbin/mount_nullfs /usr/ports /jails/haraka/usr/ports
    /sbin/mount_nullfs /var/cache/pkg /jails/haraka/var/cache/pkg
**jexec: jail "haraka" not found**
jexec: jail "haraka" not found
all done!
    /sbin/umount /jails/haraka/usr/ports
    /sbin/umount /jails/haraka/var/cache/pkg
root@smtpd:~ #

However an ls in /jails shows haraka is there:

root@smtpd:~ # cd /jails/
root@smtpd:/jails # ls
avg                     dns.last                haproxy.last            mysql.last              vpopmail
avg.last                dovecot                 **haraka**                  redis                   vpopmail.last
base-10.2-RELEASE       dovecot.last            **haraka-saved**            rspamd                  webmail
clamav                  geoip                   **haraka.last**             rspamd.last             webmail.last
config                  geoip.last              monitor                 spamassassin
dns                     haproxy                 mysql                   spamassassin.last

At this point, since there is no way to reach you in such an emergency, created new VPS and followed Mail-Toaster-6 instructions. Should mention that we rsync'd haraka/config /jails/  first so we had an intact working 'config'.

After creation of the new smtpd Mail-Toaster-6 server, we noticed right away **/jails/haraka/usr/local/haraka** does not exist. Yes, there is /data/haraka. So thinking you made some sort of change to the build of Mail-Toaster-6 as it applies to 'provision haraka' where the provision haraka script no longer creates /jails/haraka/usr/local/haraka we rsync'd our existing 'config' from old smtpd server to newly created smtpd server to /data/haraka/

Thing is, trying to get you to say what has happened to the provision haraka build with respect to /jails/haraka/usr/local/haraka not existing.
